test/test_sort_and_select.py

```diff
@@ -167,6 +167,28 @@ def test_sort_large(self, device, dtype):
         self.assertEqual(vm, torch.arange(255, dtype=dtype, device=device))
         self.assertEqual(im, t0.sort().indices)
 
+
+    @dtypes(torch.float32)
+    def test_sort_restride(self, device, dtype):
+        shape = (3, 5)
+        inputs = [
+            torch.randn(shape, dtype=dtype, device=device),
+            torch.zeros(shape, dtype=dtype, device=device),
+            torch.zeros(shape, dtype=torch.long, device=device),
+        ]
+        # Input: non-contiguous (stride: 5) 3-element array
+        tensor = inputs[0][:, 0]
+        # Outputs: non-contiguous (stride: 5) 2-element array
+        values = inputs[1][0:2, 0]
+        indices = inputs[2][0:2, 0]
```

Comment:
Hmmm.. will this test the same behavior from the original [bug](https://github.com/pytorch/pytorch/issues/65863)? I think so but I'm not sure.

In the original repro example, the two output tensors were actually too small, vs. here they are views into a larger tensor of the right size. Might be worth just making this test more similar to the repro?

Reply:
You are right. I did it like this because I wanted to make it explicit that the strides weren't `(1,)` before. But, I guess that's not really necessary. Will change it.